controller/src/job/mod.rs

```diff
@@ -109,7 +109,7 @@ pub(crate) async fn delete_job(k8s_client: kube::Client, name: &str) -> JobResul
             &DeleteParams {
                 dry_run: false,
                 grace_period_seconds: Some(0),
-                propagation_policy: Some(PropagationPolicy::Foreground),
+                propagation_policy: Some(PropagationPolicy::Background),
```

Comment:
Does this mean that the `Resource` will be deleted as soon as the delete starts? If so, it may become problematic.

Reply:
No this is for deletion of the Job (i.e. the Pod). So once a test agent or resource agent is already running, and we want to delete that job (i.e. after it's work is done and we are just terminating it), we are allowing that pod termination to happen in the background.